charms/istio-pilot/src/resources_handler.py

```diff
@@ -72,31 +76,55 @@ def delete_existing_resources(
                 obj,
                 namespace=namespace,
                 ignore_not_found=ignore_not_found,
-                ignore_unauthorized=ignore_unauthorized,
             )
 
     def apply_manifest(self, manifest, namespace=None):
         for obj in codecs.load_all_yaml(manifest):
             self.lightkube_client.apply(obj, namespace=namespace)
 
     def delete_manifest(
-        self, manifest, namespace=None, ignore_not_found=False, ignore_unauthorized=False
+        self,
+        manifest,
+        namespace=None,
+        ignore_not_found=False,
+        include_created_by_external=False,
     ):
-        for obj in codecs.load_all_yaml(manifest):
-            self.delete_resource(
-                obj,
-                namespace=namespace,
-                ignore_not_found=ignore_not_found,
-                ignore_unauthorized=ignore_unauthorized,
-            )
-
-    def generate_generic_resource_class(self, filename: str) -> GenericNamespacedResource:
+        for obj in yaml.safe_load_all(manifest):
+            if not obj:
+                self.log.warning(
+                    "Attempting to delete a resource from an empty resource object.\n"
+                    "This could be due to an incorrectly formatted manifest."
+                )
+                return
+            try:
+                obj_to_delete = lightkube.codecs.from_dict(obj)
+            except LoadResourceError:
+                if include_created_by_external:
+                    self.generate_generic_resource_class(manifest=obj)
+                    obj_to_delete = lightkube.codecs.from_dict(obj)
+                else:
+                    raise
```

Comment:
Can you explain what this does?  I'm not sure what the `LoadResourceError` would be indicating.